packages/services/src/view-service/status-stream.ts

```diff
@@ -4,18 +4,14 @@ import { servicesCtx } from '../ctx/prax';
 export const statusStream: Impl['statusStream'] = async function* (_, ctx) {
   const services = await ctx.values.get(servicesCtx)();
   const { indexedDb } = await services.getWalletServices();
-  const latestBlockHeight = await services.querier.tendermint.latestBlockHeight();
 
-  // As syncing does not end, nor does this stream.
-  // It waits for events triggered externally when block sync has progressed.
-  const subscription = indexedDb.subscribe('FULL_SYNC_HEIGHT');
-
-  for await (const update of subscription) {
-    const syncHeight = update.value;
-    yield {
-      latestKnownBlockHeight: syncHeight >= latestBlockHeight ? syncHeight : latestBlockHeight,
-      partialSyncHeight: syncHeight,
-      fullSyncHeight: syncHeight,
-    };
-  }
+  // This should stream forever unless cancelled.
+  let remoteBlockHeight: bigint | undefined;
+  for await (const { value: syncHeight } of indexedDb.subscribe('FULL_SYNC_HEIGHT'))
+    if ((remoteBlockHeight ??= await services.querier.tendermint.latestBlockHeight()))
+      yield {
+        latestKnownBlockHeight: syncHeight <= remoteBlockHeight ? remoteBlockHeight : syncHeight,
+        partialSyncHeight: syncHeight,
+        fullSyncHeight: syncHeight,
+      };
```

Comment:
The lack of brackets make this hard to understand. Don't think we should ever do bracket-less loops.